src/AnalysisEcce.cxx

```diff
@@ -205,7 +215,32 @@ void AnalysisEcce::Execute()
         }
       }
     } // end truth loop
+    
+    for(int imc=0; imc<hepmcp_PDG.GetSize(); imc++) {
+      int pid_ = hepmcp_PDG[imc];     
+      int genStatus_ = hepmcp_status[imc]; // genStatus 4: beam particle,  1: final state                                                                            						
 
+      double px_ = hepmcp_psx[imc];
+      double py_ = hepmcp_psy[imc];
+      double pz_ = hepmcp_psz[imc];
+      double e_  = hepmcp_E[imc];
+      
+      double p_ = sqrt(pow(hepmcp_psx[imc],2) + pow(hepmcp_psy[imc],2) + pow(hepmcp_psz[imc],2));
+      double mass_ = (fabs(pid_)==211)?pimass:(fabs(pid_)==321)?kmass:(fabs(pid_)==11)?emass:(fabs(pid_)==13)?mumass:(fabs(pid_)==2212)?pmass:0.;      
+      // add to `mcpart`                                                                                                                                                                                                                      
+	   int mother = hepmcp_m1[imc];
+      if(pid_ == 22 || pid_ == 23){
+	if(genStatus_ == 1){
+	  if( mother == genEleBCID ){
+	    TLorentzVector FSRmom(px_, py_, pz_, e_);
+	    TLorentzVector eleCorr = kinTrue->vecElectron + FSRmom;
+	    kinTrue->vecElectron.SetPxPyPzE(eleCorr.Px(), eleCorr.Py(), eleCorr.Pz(), eleCorr.E());
+	  }
+	}
+      }
+      
+      
+    }
```

Comment:
Add a comment to explain what this loop does. Seems like we will want this in the other `Analysis`-derived classes, right?

Reply:
Should be included in other classes yes, will try to update this for at least EPIC and Delphes classes soon, where we're more likely to have data generated with QED radiation.